src/client/resources/common.ts

```diff
@@ -50,3 +56,41 @@ export abstract class ClusterResource {
 function namedPath(listPath, name) {
   return [listPath, name].join('/');
 }
+
+export abstract class ClusterDiscoveryResource {
+
+  private _type: string;
+  private _cluster: string;
+
+  public abstract discoveryParameters(): IDiscoveryParametrs;
+  public abstract for(): string;
+
+  constructor(cluster: string, type: string) {
+    if (!cluster) {
+      throw new Error('ClusterDiscoveryResource must have a cluster, it was undefined');
+    }
+    this._cluster = cluster;
+    this._type = type;
+  }
+
+  public discoveryType() { return this._type + '/'; }
+
+  public discoveryCluster() {
+    return ['clusters', this._cluster].join('/');
+  }
+
+  public parametrizedPath(params = {}) {
+    if (this.discoveryParameters().wait !== -1) {
```

Comment:
RC: `-1` here is a [magic number](https://en.wikipedia.org/wiki/Magic_number_(programming)) and an anti-pattern. It would be better to define a const somewhere like `UNSPECIFIED_DISCOVERY_PARAM`. Could even go as far as to typedef the int, but I don't think that's necessary.